docs/mobile-apps/automated-testing/appium/appium-migration.md

```diff
@@ -0,0 +1,163 @@
+
+ Throughout your test files, replace your TestObject credentials with your Sauce Labs credentials.
+
+  We recommend using [Environment Variables for Authentication](https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Use+Environment+Variables+for+Authentication+Credentials) in order to reduce hard coding and protect your personal information from exposure.
+
+
+### Step 2: Replace Remote URL Endpoint
+
+Replace the Remote URL endpoint in your tests to access the Sauce Labs platform, for example:
+
+|Region|Test Object URL|Sauce Labs URL|
+|---|-----------|-----------|
+|US|<small>`https://us1.appium.testobject.com/wd/hub`</small>|<small>`https://ondemand.us-west-1.saucelabs.com/wd/hub`</small>|
+|EU|<small>`https://eu1.appium.testobject.com/wd/hub`</small>|<small>`https://ondemand.eu-central-1.saucelabs.com/wd/hub`</small>|
+
+If you're using [Sauce Connect Proxy](/secure-connections/sauce-connect), our [REST API](/dev/api), or any other features in conjunction with real device testing, consult the documentation for that feature for applicable Sauce Labs endpoints.
+
+
+### Step 3: Change Device Test Options
+
+Not all device model and operating system combinations will carry over from the TestObject platform to Sauce Labs. You may need to adjust your device capabilities in your test configuration.
+
+#### **Device Allocations**
+To ensure your preferred [device allocations](https://wiki.saucelabs.com/display/DOCS/Mobile+Application+Testing+Admin+Guide) will still work when you move from TestObject to Sauce Labs, view the list of available devices for your region by logging into the Sauce Labs platform and going to **Live** > **Cross-Browser** > **Mobile Real**. For a full, non-region specific list of supported devices and platforms, see [Supported Browsers and Devices](https://saucelabs.com/platform/supported-browsers-devices).
+
+If you switch your Sauce Labs Data Center in the UI, you may see a different set of devices. Each Data Center stores a different set of mobile real devices.
+
+#### **Appium Capabilities**
+
+Validate that the Appium capabilities you've used on TestObject will still work when you move tests to Sauce Labs; see [Appium Capabilities for Real Device Testing](/basics/test-config-annotation/test-config).
+
+### Step 4: Upload your Application to Sauce Application Storage
+
+Any mobile apps in TestObject storage will no longer be accessible after TestObject reaches end-of-life. Please upload these to Sauce Application Storage.
+
+*   For instructions on uploading your mobile apps (.apk or .ipa files) to Sauce Labs, see [Application Storage](/mobile-apps/app-storage).
+*   Update all app references in your test code capabilities. For example, in the `app` capability, you must use `storage:<app-id>` or `storage:filename=<some-app.apk>`, rather than `sauce-storage:<myapp.apk>` as you would in TestObject. See [Using Appium for Automated Mobile App Testing](/mobile-apps/appium) for more information.
```

Comment:
```suggestion
*   Update all app references in your test code capabilities. For example, in the `app` capability, you must use `storage:<app-id>` or `storage:filename=<some-app.apk>`, rather than `sauce-storage:<myapp.apk>` as you would in TestObject. See [Using Appium for Automated Mobile App Testing](/mobile-apps/automated-testing/appium) for more information.
```